src/lib/int.ts

```diff
@@ -19,23 +19,41 @@ class UInt64 extends CircuitValue {
   @prop value: Field;
   static NUM_BITS = 64;
 
-  constructor(x: UInt64 | UInt32 | Field | number | string | bigint) {
-    if (x instanceof UInt64 || x instanceof UInt32) x = x.value;
-    else if (!(x instanceof Field)) x = Field(x);
-    super(x);
+  /**
+   * Create a {@link UInt64}.
+   * The max value of a {@link UInt64} is `2^64 - 1 = UInt64.MAXINT()`.
+   *
+   * **Warning**: Cannot overflow, an error is thrown if the result is greater than UInt64.MAXINT()
+   */
+  constructor(x: UInt64 | UInt32 | FieldVar | number | string | bigint) {
+    if (x instanceof UInt64 || x instanceof UInt32) x = x.value.value;
+    super({ value: Field(x) });
+    UInt64.checkConstant(this.value);
```

Comment:
this doesn't seem to work, it throws an error in every CI test
```suggestion
    let value = Field(x);
    super({ value });
    UInt64.checkConstant(value);
```